pkg/uevent/listener.go

```diff
@@ -161,11 +294,11 @@ func (l *listener) msgRead(buf *[]byte) error {
 
 // ReadMsg allow to read an entire uevent msg
 func (l *listener) ReadMsg() ([]byte, error) {
-	_, buf, err := c.msgPeek()
+	_, buf, err := l.msgPeek()
 	if err != nil {
 		return nil, err
 	}
-	if err = c.msgRead(buf); err != nil {
+	if err = l.msgRead(buf); err != nil {
```

Comment:
In `msgRead()` error out after `syscall.Recvfrom()` if `n != len(buf)`. This never happens though.